src/cljs/org/broadinstitute/firecloud_ui/common/table.cljs

```diff
@@ -329,60 +358,61 @@
                                         (func type))))
-                :data entity-list
-                :->row (if count-column
-                         (fn [m]
-                           (vec (concat [m]
-                                        [(count-data m)]
-                                        (map #(get-in m ["attributes" %]) attribute-keys))))
-                         (fn [m]
-                           (vec (concat [m]
-                                        (map #(get-in m ["attributes" %]) attribute-keys)))))})]))]))
+                :->row (fn [m]
+                         (->> attributes (map #(get-in m ["attributes" %])) (cons m) vec))})]))]))
    :component-did-mount
-   (fn [{:keys [props state this]}]
+   (fn [{:keys [props state]}]
      (endpoints/call-ajax-orch
        {:endpoint (endpoints/get-entity-types (:workspace-id props))
         :on-done (fn [{:keys [success? get-parsed-response]}]
                    (if success?
-                     (let [types (vec (get-parsed-response))
-                           first-type (or (:initial-entity-type props) (first (first types)))]
+                     (let [metadata (get-parsed-response)
+                           entity-types (vec (keys metadata))]
                        (swap! state update-in [:server-response]
-                         assoc :entity-types types :selected-entity-type first-type)
-                       (react/call :load-type this first-type))
+                         assoc :entity-metadata metadata
+                               :entity-types entity-types
+                               :selected-entity-type (or (:initial-entity-type props) (first entity-types))))
                      (swap! state update-in [:server-response]
                        assoc :server-error (get-parsed-response))))}))
-   :load-type
-   (fn [{:keys [props state]} type]
-     (if-not type
-       (swap! state update-in [:server-response] assoc :entity-list [])
-       (do
-         (swap! state assoc :loading-entities? true)
-         (endpoints/call-ajax-orch
-           {:endpoint (endpoints/get-entities-of-type (:workspace-id props) type)
-            :on-done (fn [{:keys [success? get-parsed-response]}]
-                       (swap! state update-in [:server-response]
-                         assoc (if success? :entity-list :server-error) (get-parsed-response))
-                       (swap! state dissoc :loading-entities?)
-                       (when-let [func (:on-filter-change props)]
-                         (func type)))}))))})
+   :pagination
+   (fn [{:keys [props state]} columns]
+     (let [{{:keys [entity-types]} :server-response} @state]
+       (fn [{:keys [current-page rows-per-page filter-text sort-column sort-order filter-group-index]} callback]
+         (let [type (nth entity-types filter-group-index)
+               sort-field (if (zero? sort-column)
+                            "name"
```

Comment:
is this the only place where the UI provides a default value? Otherwise, we use the defaults provided by the API? I'd like to make it consistent - all API defaults or all UI defaults, not mixed.